File: Array & string/checkpermutation.java
import java.util.*;
public class checkpermutation {
    public static void main(String[] args) {
        System.out.println(check("dog","god"));
    }
    public static boolean check(String s,String t){
        if(s.length()!=t.length()){
            return false;
        }
        char[] c1=s.toCharArray();
        Arrays.sort(c1);
        String s1=Arrays.toString(c1);
        char[] c2=t.toCharArray();
        Arrays.sort(c2);
        String t1=Arrays.toString(c2);
        if(s1.equals(t1)){
            return true;
        }
        return false;
    }
}

File: Array & string/checkpermutation.cpp
#include<bits/stdc++.h>
using namespace std;

bool check(string s,string t){
    if(t.length()!=s.length()){
        return false;
    }
    sort(s.begin(),s.end());
    sort(t.begin(),t.end());
    if(s.compare(t) == 0){
        return true;
    }
    return false;
}
int main(){
    cout << check("doge","god") << "\n";
    string r = "rohit";
    cout << r[3] << "\n";
    return 0;
}